### DISCUSSION TITLE
Armbian images with reticulum preinstalled

### DISCUSSION THREAD

**Discussion by Vault2501** [markqvist/Reticulum#97] [Show and tell]
EDIT: 
I made a page to link to built images and documentation, will try to keep them up to date: https://github.com/Vault2501/Meshed-Banana-image/wiki/Images

Hi Mark and all, I was working on a project I called Meshed Banana some time ago, but had to stop for personal reasons. 
I also remember this was mentioned in some thread here, but could not find it, so I'll just start a fresh one.

I did pick up the pieces recently, and am about to get it finally properly documented, So, currently, I have the scripts I use to create Armbian images with reticulum preinstalled in a repo at  https://github.com/Vault2501/Meshed-Banana-image
I tested the image building with the latest Armbian build system, and documented in the README on how to ue the scripts with it. 

However, you can also run the scripts on an installed debian like system to install reticulum with one step. I'm still testing this, so maybe don't try it yet yourself ;)

Note: This won't install reticulum as root, but create a user installation, moves that to '/opt' and then modifies the /etc/skel/.profile so that newly created users get it copied into their home upon first login. So it's a bit around the corner ;)

**Comment by markqvist** [comment]
Very cool to hear you are back working on that project, and thanks for letting us all know! I am going to try it out soon so I can provide feedback, but currently wrestling with some annoying build issues for Sideband, and need to get all of that out of the way first ;)

**Reply by Vault2501** [reply]
Ok, I did rework and move my documenatation. It;s not fully complete, some  pictures are still missing, but nearly everything is in place. Still editing an assembly video, that will be added once done.

https://github.com/Vault2501/Meshed-Banana-Device

**Comment by Vault2501** [comment]
Still working on the documentation, just tested that the script also can be run stand alone on a booted system, seems to work so for. I did some small changes, so will run a complete test of the documentaion later today, and then also make an image available. Once that is done, I'll add the whole hardware part, so that this can be used by whoever is interested.

**Comment by 4c3e** [comment]
This is great! I should be able to test this on a pi soon. Thanks for your work :smile:

**Comment by Vault2501** [comment]
I think I'll move some of the documentation to the github wiki of the repo soon, then I can structure it in a way that one only needs to read what he requires. I'll add a section on how to build for RPi4 there as well, as it's the most common SBC type out there. Please note, afair Armbian only supports RPi4 models, nothing prior. But not 100% sure.

**Reply by markqvist** [reply]
This would be great too!

**Reply by Vault2501** [reply]
Moved it and created https://github.com/Vault2501/Meshed-Banana-image/wiki/Build-RPi4-Image
I'm currently building an rpi4 image, seems that only Ubuntu releases are supported. I might also switch to that with the Meshed Banana, to keep it consistent.
Once the build here succeeds, I'll put a link to the image in the documentation, so someone can test it, as I don;t have a rpi4 available for testing here atm.

**Comment by Vault2501** [comment]
I added the link to an image I just built for the Raspberry Pi 4b. As I don't have one, I have no idea if it works, but if you don't want to try building yourself, you can give it a boot: https://drive.google.com/file/d/1tp8m_ybmCyZzQIcNgD8Dr9ulk2z61yER/view?usp=sharing

I think the biggest issue would be the pre installed config - that most likely needs to be adjusted. But reticulum should be available to newly created users: https://github.com/Vault2501/Meshed-Banana-image/wiki/Build-RPi4-Image#creating-new-users

**Reply by Vault2501** [reply]
Removed the image, I'm not sure, but I think a disk space issue caused it to be truncated. I'm building a new one later today, with xfce desktop and sideband installed, just first testing on the Meshed Banana to make sure all works. Then I'll add a new link

**Reply by Vault2501** [reply]
Added image again, only a minimal build as I'm still having trouble bulding with a desktop.

**Comment by Vault2501** [comment]
Small update - managed to setup one Meshed Banana as access point, and connected an android phone running sideband to it. Connected another Meshed Banana to my home wifi, and started sideband on a linux desktop, also in the home wifi. Tested sending messages propagated and direct from the desktop to the andoid phone. Both worked like a charm. Even when I went out of reach, and then came back into reach, messages arrive at the destination. Will later try to build a third Meshed Banana and then see if the meshing works. Also will add the configs for access point mode to the image, so one can switch between client and ap mode.

**Reply by markqvist** [reply]
This is so good to hear. Really digging how this is coming together.

The next version of Sideband should play really well with this. In this version, Reticulum and the LXMF router will be launched as a proper service on Android, and it will also support notifications. Identity import and export is implemented as well. So we are getting pretty close to something that is really useful.

**Reply by Vault2501** [reply]
I tweaked the images some more, now I have a user preinstalled and rnsd started as service. Also now have an image that starts by default as access point, so one can simply turn it on, and it routes between wifi and lora, currently testing this atm and updating the docs accordingly. When this all works as simple as possible, I'll create a version 0.1 ;)
From that on I'll continue more organized, like work in branches to add new features, like a desktop with vnc access and config tool...

**Comment by samuk** [comment]
Could this be made to work with a Mango Pi MQ?

It's a small  open hardware RiscV board: 
https://mangopi.org/mangopi_mq
https://github.com/mangopi-sbc/MQ

Which runs Tina Linux: https://github.com/mangopi-sbc/Tina-Linux#readme

Or perhaps this similar board https://hackaday.io/project/182389-the-cheapest-risc-v-64-computer-by-now

It would then be possible to fabricate a carrier to connect a couple of LoRa modules https://www.olimex.com/Products/IoT/LoRa/LoRa868/open-source-hardware
